src/cascade/dismod/serialize.py

```diff
@@ -0,0 +1,327 @@
+        "parent": np.array([np.NaN]),
+    })
+    bundle_fit.node = node_table
+
+    non_zero_rates = list(model.smoothers.keys())
+    if 'iota' in non_zero_rates:
+        if 'rho' in non_zero_rates:
+            value = 'iota_pos_rho_pos'
+        else:
+            value = 'iota_pos_rho_zero'
+    else:
+        if 'rho' in non_zero_rates:
+            value = 'iota_zero_rho_pos'
+        else:
+            value = 'iota_zero_rho_zero'
+    bundle_fit.option = pd.DataFrame({
+        "option_name": ["parent_node_name", "rate_case"],
+        "option_value": [node_table.iloc[0]["node_name"], value],
+    })
+
+    observations = observations_to_data(model.observations)
+    constraints = observations_to_data(model.constraints, hold_out=1)
+    total_data = pd.concat([observations, constraints], ignore_index=True)
+    # Why a unique string name?
+    total_data["data_name"] = total_data.index.astype(str)
+    bundle_fit.data = total_data
+
+    # Include all data in the data_subset.
+    bundle_fit.data_subset = pd.DataFrame({
+        "data_id": np.arange(len(total_data)),
+    })
+
+    # Ages and times are used by Weight grids and smooth grids,
+    # so pull all ages and times from those two objects in the
+    # internal model. Skip weight grid here b/c assuming use constant.
+    bundle_fit.age, bundle_fit.time = age_time_from_grids(
+        model.smoothers.values(), total_data)
+
+    # These are used to get the age_id into another dataframe.
+    # Use pd.merge_asof to do this.
+    age_df = bundle_fit.age.copy()
+    age_df["age_id"] = age_df.index
+    time_df = bundle_fit.time.copy()
+    time_df["time_id"] = time_df.index
+
+    bundle_fit.weight, bundle_fit.weight_grid = simplest_weight()
+
+    bundle_fit.prior = model.priors
+    prior_df = bundle_fit.prior.copy()
+    prior_df["prior_id"] = prior_df.index
+
+    # The avgint needs to be translated.
+    bundle_fit.avgint = pd.DataFrame({
+        "integrand_id": model.outputs.integrand.apply(lambda x: x.value),
+        # Assumes one location_id.
+        "node_id": 0,
+        # Assuming using the first set of weights, which is constant.
+        "weight_id": 0,
+        "age_lower": model.outputs.age_start,
+        "age_upper": model.outputs.age_end,
+        "time_lower": model.outputs.year_start,
+        "time_upper": model.outputs.year_end,
+    })
+
+    bundle_fit.smooth, bundle_fit.smooth_grid = convert_smoothers(
+        model.smoothers, age_df, time_df, prior_df)
+    smooth_df = bundle_fit.smooth.copy()
+    smooth_df["smooth_id"] = smooth_df.index
+
+    rate_enum = enum_to_dataframe(RateName)
+    with_smooth = rate_enum.merge(smooth_df, left_on="name", right_on="smooth_name", how="outer")
+    bundle_fit.rate = pd.DataFrame({
+        "rate_name": with_smooth["name"],
+        "parent_smooth_id": with_smooth["smooth_id"],
+        "child_smooth_id": np.NaN,
+        "child_nslist_id": np.NaN,
+    })
+
+    flush_begin = timer()
+    bundle_fit.flush()
```

Comment:
Return this guy instead of flushing so the caller has more control.

Reply:
done